File: app/src/test/java/kztproject/jp/splacounter/DummyCreator.java
package kztproject.jp.splacounter;

import kztproject.jp.splacounter.model.Counter;
import kztproject.jp.splacounter.model.User;
import kztproject.jp.splacounter.model.UserResponse;

/**
 * Created by k-seito on 2017/07/19.
 */

public class DummyCreator {

    public static UserResponse createDummyUserResponse() {
        UserResponse response = new UserResponse();
        User user = new User();
        user.setId(1);
        user.setFullName("test_user");
        response.setUser(user);
        return response;
    }

    public static Counter createDummyCounter() {
        Counter counter = new Counter();
        counter.setId(1);
        counter.setCount(10);
        return counter;
    }
}

File: app/src/test/java/kztproject/jp/splacounter/DummyCreator.kt
package kztproject.jp.splacounter

import kztproject.jp.splacounter.auth.api.model.RewardUser
import kztproject.jp.splacounter.auth.api.model.TodoistUser
import kztproject.jp.splacounter.auth.api.model.UserResponse
import kztproject.jp.splacounter.reward.database.model.Reward
import kztproject.jp.splacounter.reward.database.model.RewardPoint

object DummyCreator {

    fun createDummyUserResponse(): UserResponse {
        val response = UserResponse()
        val user = TodoistUser()
        user.id = 1
        user.fullName = "test_user"
        response.user = user
        return response
    }

    fun createDummyRewardUser(): RewardUser {
        return RewardUser(10, 123, 0)
    }

    fun createDummyReward(): Reward {
        return Reward(1, "Test", 5, "Test description", true)
    }

    fun createDummyNoRepeatReward(): Reward {
        return Reward(2, "Test2", 7, "Test description", false)
    }

    fun createDummyRewardPoint(): RewardPoint {
        return RewardPoint(10)
    }
}
